### PR TITLE
Add Jam

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
does anyone @conda-forge/help-c-cpp have any hint about this error here?

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=42126&view=logs&jobId=10bda42a-d420-5c28-a43f-bc4e66405e5c&taskId=bef775d0-a1a2-55a3-9721-17147029baf0&lineStart=607&lineEnd=607&colStart=1&colEnd=6

**Comment by marcelotrevisani** [comment]
@conda-forge/staged-recipes please review

File: recipes/jam/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  sha256: 7c510be24dc9d0912886c4364dc17a013e042408386f6b937e30bd9928d5223c
+  # Removed CC and CFLAG in Makefile 
+  patches: remove_cc.patch
+
+build:
+  number: 0
+  skip: True  # [win]
+  script:
+    - export AR="ar r"  # [unix]
+    - make -j${CPU_COUNT} VERBOSE=1 CC="$CC" CXX="$CXX" CCFLAGS="$CFLAGS" RANLIB="$RANLIB" LINKFLAGS="$LDFLAGS" Cc="${CC}" --environment-overrides BINDIR=$PREFIX/bin   # [unix]
```
Comment (chrisburr) [inline recipes/jam/meta.yaml]: Is `Cc` a typo or intentional?

File: recipes/jam/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+
+test:
+  commands:
+    - jam -v
+
+about:
+  home: https://swarm.workshop.perforce.com/files/guest/perforce_software/jam
+  license: BSD-2.0-Clause
```
Comment (chrisburr) [inline recipes/jam/meta.yaml]:
```suggestion
  license: BSD-2-Clause
```

File: recipes/jam/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  url: https://swarm.workshop.perforce.com/downloads/guest/perforce_software/jam/jam-{{ version }}.zip
+  sha256: 7c510be24dc9d0912886c4364dc17a013e042408386f6b937e30bd9928d5223c
+  # Removed CC and CFLAG in Makefile 
+  patches: remove_cc.patch
+
+build:
+  number: 0
+  skip: True  # [win]
+  script:
+    - export AR="ar r"  # [unix]
```
Comment (chrisburr) [inline recipes/jam/meta.yaml]: This is using `ar` from the host. I think this works:
```suggestion
    - export AR="${AR} r"  # [unix]
```

File: recipes/jam/meta.yaml (hunk 1 above)
Comment (marcelotrevisani) [reply recipes/jam/meta.yaml]: It is not necessary anymore, it was just for testing purposes, I am going to delete it, thanks for spotting that! :)

File: recipes/jam/meta.yaml (hunk 3 above)
Comment (marcelotrevisani) [reply recipes/jam/meta.yaml]: Indeed! My mistake, thanks!

**Pull request merged**